src/main/scala/akka/persistence/cassandra/ResultSetActorPublisher.scala

```diff
@@ -26,20 +32,29 @@ object ResultSetActorPublisher {
  * by invoking Source.actorPublisher(...).
  */
 class ResultSetActorPublisher[T](
-    resultSetFuture: Future[ResultSet], implicit val rowMapper: RowMapper[T]
+    resultSetFuture: ResultSetFuture,
+    config: CassandraReadJournalConfig,
+    implicit val rowMapper: RowMapper[T]
 ) extends ActorPublisher[T] with Stash with ActorLogging {
 
   import context.dispatcher
+  val execTimeout = Duration(config.execTimeout.toMillis(), TimeUnit.MILLISECONDS)
 
   resultSetFuture map ResultSetReady pipeTo self
+  val timeoutSchedule = context.system.scheduler.scheduleOnce(execTimeout, self, ExecTimeout)
 
   def receive = {
     case ResultSetReady(resultSet) =>
+      timeoutSchedule.cancel()
       log.debug("Resultset is ready with {} available results, exhausted: {}",
           resultSet.getAvailableWithoutFetching, resultSet.isExhausted)
       context become ready(resultSet)
       deliver(resultSet)
       unstashAll()
+      
+    case ExecTimeout =>
+      resultSetFuture.cancel(true)
```

Comment:
consider using named parameter, so it's more clear what boolean value stands for.